lib/ansible/module_utils/common/sys_info.py

```diff
@@ -137,7 +137,7 @@ def __new__(cls, args, kwargs):
         class User:
             def __new__(cls, args, kwargs):
                 new_cls = get_platform_subclass(User)
-                return super(cls, new_cls).__new__(new_cls, args, kwargs)
+                return super(cls, new_cls).__new__(new_cls)
```

Comment:
I don't believe this is correct. `args` and `kwargs` need to be passed to class when it is instantiated. This should probably be `*args, **kwargs`, though.

Reply:
I know this is what it done in `load_platform_subclass()` but that looks incorrect as well.

Reply:
Some tests were failing complaining about this.

Reply:
```/ansible_user_payload_r6aajx5l/ansible_user_payload.zip/ansible/modules/system/user.py\", line 2989, in <module>\n  File \"/tmp/ansible_user_payload_r6aajx5l/ansible_user_payload.zip/ansible/modules/system/user.py\", line 2877, in main\n  File \"/tmp/ansible_user_payload_r6aajx5l/ansible_user_payload.zip/ansible/modules/system/user.py\", line 463, in __new__\nTypeError: object.__new__() takes exactly one argument (the type to instantiate)\n",```

Reply:
I'll have to look into this more. I'll find out more tomorrow when the author gets back from PTO.

Reply:
Alright, after a deep dive into what `__new__()` is actually doing, this change is correct.

`__new__()` happens before `__init__()` and simply returns the instantiated but not initialized object. `args, kwargs` would be passed to `__new__()` in order to affect the object selection logic. They would not affect object initialization.

Since they are not used by our `__new__()` method, they can be omitted from the `super().__new_()` call.